src/librustc/infer/error_reporting/mod.rs

```diff
@@ -1299,6 +1302,117 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
             _ => {}
         }
 
+        #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
```

Comment:
Can you add a comment re. the role of this type (maybe clarify the distinction with the "real" `TyKind`)

Reply:
Also I wonder if it should be slightly renamed to avoid confusion with the `ty::TyKind`?